eval/mt_bench/components.py

```diff
@@ -0,0 +1,99 @@
+# type: ignore
+# pylint: disable=no-value-for-parameter,import-outside-toplevel,import-error
+from typing import List, NamedTuple
+from kfp.dsl import component, Input, Output, Artifact, Model, importer
+from utils.consts import TOOLBOX_IMAGE
+
+INSTRUCTLAB_IMAGE = "registry.redhat.io/rhelai1/instructlab-nvidia-rhel9:1.1"
+
+@component(base_image=TOOLBOX_IMAGE)
+def parse_output_to_dict(output: str):
+    lines = output.strip().split('\n')
+    report = {}
+    
+    for line in lines:
+        line = line.strip()
+        if line.startswith("## MODEL"):
+            report["model"] = lines[lines.index(line) + 1].strip()
+        elif line.startswith("### AVERAGE"):
+            average_line = lines[lines.index(line) + 1].strip()
+            report["average"] = float(average_line.split()[0])
+            report["average_details"] = average_line
+        elif line.startswith("### TURN ONE"):
+            report["turn_one"] = float(lines[lines.index(line) + 1].strip())
+        elif line.startswith("### TURN TWO"):
+            report["turn_two"] = float(lines[lines.index(line) + 1].strip())
+        elif line.startswith("### ERROR RATE"):
+            report["error_rate"] = float(lines[lines.index(line) + 1].strip())
+    
+    return report
+
+
+@component(base_image=INSTRUCTLAB_IMAGE)
+def run_mt_bench_op(
+    mt_bench_output: Output[Artifact],
+    models_path_prefix: str,
+    judge_model_path: str,
+    models_list: List[int],
+) -> NamedTuple("outputs", best_model=str, best_score=float) :
+    import json
+    import os
+    import subprocess
+    import typing
+
+    Outputs = NamedTuple("outputs", best_model=str, best_score=float)
+    os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
+
+    # TODO: use external model after adding option to CLI
+    # ilab CLI does not support external models
+    #judge_model_name = os.getenv("JUDGE_MODEL_NAME")
+    #judge_endpoint = os.getenv("JUDGE_ENDPOINT")
+    #judge_api_key = os.getenv("JUDGE_API_KEY", "")
+
+    scores = {}
+    all_mt_bench_data = []
+    for model_name in models_list:
+        model_path = f"{models_path_prefix}/{model_name}"
+
+        #ilab model evaluate --benchmark mt_bench
+        #    --model $ILAB_MODELS_DIR/granite-7b-lab-Q4_K_M.gguf
+        #    --judge-model $ILAB_MODELS_DIR/granite-7b-lab-Q4_K_M.gguf
+
+        command = ["ilab", "model", "evaluate", "--benchmark", "mt_bench", "--model", "{models_path_prefix}/{model_name}",  "--judge-model", "{judge_model_path}"]
+        result = subprocess.run(command, capture_output=True, text=True)
```

Comment:
If you are going to use the CLI, it might work to change this from a component to a container_component? Something like:

```
@dsl.container_component
def run_mt_bench_op(
    mt_bench_output: Output[Artifact],
    models_path_prefix: str,
    judge_model_path: str,
    models_list: List[int],
    ) -> NamedTuple("outputs", best_model=str, best_score=float) :
    
    return dsl.ContainerSpec(
        INSTRUCTLAB_IMAGE,
        ["/bin/sh", "-c"],
        [ f"ilab model evaluate --benchmark mt_bench --model {models_path_prefix}/{model_name} --judge-model {judge_model_path}"
        ],
    )
```

Reply:
will do - i was not sure if this was the approach we were going to go with, or keep with using the libraries like what I have going on[ here](https://github.com/redhat-et/ilab-on-ocp/blob/main/eval/mt_bench_pipeline.py)  - so was mostly putting this up to see if it would run.